### ISSUE TITLE
GeyserMC crashing the app/game without errors when switching server too fast on Bungeecord

### ISSUE THREAD

**Issue by Leonardo-Fiori** [GeyserMC/Geyser#2669]
### Describe the bug

My GeyserMC setup has worked flawlessy for an entier year. Some months ago I introduced a queue system that is a server where players connect to. Once it's their turn, they get sent to the main server. 

If there is empty space in the servers slots the queue sends the player instantly to the main server: in this moment the players game crashes without error messages. The app instantly closes.

This has never happened before upgrading to 1.18 Geyser Builds.

I am running latest bungeecord build, but geysermc is standalone on a different machine.

### To Reproduce

Send players instantly to another server as soon as they join the default server. If the switch happens too fast, their game will crash.

This happens on:
iPad Mini 4th gen
iPad 6th gen
iPad Air 2
Windows 10

### Expected behaviour

Game not crashing

### Screenshots / Videos

https://cdn.discordapp.com/attachments/916622439436398612/916630845710811186/video0.mov

https://cdn.discordapp.com/attachments/916622439436398612/916630678345515048/CrashingLikeacar.MP4.mp4

http://www.youtube.com/watch?v=I4hf1LfLH4Q

### Server Version and Plugins

Geyser standalone, console commands do not work. It's a build from today (04/12/2021)

Server runs on latest stable release of Yatopia, probably the only one in existance that is stable. I am locked to this version due to Java 16 issues, and I am planning to update to 1.17.1 purpur ASAP. It has been running on this version since ages, and never had a single issue.

Again, don't immediatly point it out to this, since this issue has never existed before updating to 1.18 geyser builds. I know, I know, I should probably feel ashamed to run this version lol

Yatopia version git-Yatopia-ver/1.16.5  (MC: 1.16.5)

### Geyser Version

https://dump.geysermc.org/PcZuUskJBE9AQ5zmSKPi2k5loTJOeZue

**Comment by Camotoy** [comment]
Can you explain how one of us could replicate this queue system?

**Comment by Leonardo-Fiori** [comment]
Setup a bungee network with two servers:

1. as soon as the player joins the default/forced server (queue)
2. /send the player to the other server (main server)

my queue plugin listens for the join event, and moves the player to the other server almost instantly, while the chunks of the queue are still loading

don't know if you can replicate that by hand, best trick would be to spam the /send command or code a little script that does that for you on join :)

it has worked flawlessy since months, now 15/20 bedrock players are all crashing straight on join

**Comment by Camotoy** [comment]
I'd love a pre-made snippet or code that verifiably crashes the client - I want to look into fixing this but don't want to attempt to try to replicate your setup.

**Comment by MRAsshole** [comment]
i have the same issue after upgraded geyser 1.18, hope i'll be fixed soon

**Comment by Camotoy** [comment]
If you want it fixed, please provide replication steps!!

**Comment by Leonardo-Fiori** [comment]
My dev is working on a small plugin for you to replicate the issue. As of now all I can suggest to replicate is:

Use a device from this list:
iPad Mini 4th gen
iPad 6th gen
iPad Air 2

And instantly send an user to a different sever in the bungee network as soon as they join the predefined/forced one

**Comment by Camotoy** [comment]
I don't have access to an IPad, but I will eagerly await the test plugin.

**Comment by Jens-Co** [comment]
Is this issue still going on? If so please provide us with the plugin that your dev created. Otherwise I'll close this issue if no response is given.

**Comment by Leonardo-Fiori** [comment]
Unfortunately our dev has been extremely busy. We have fixed the issue by adding a delay on server switch to let the players fully load the map before switching server. The map loading was what caused the crash. Not an elegant solution, but who knows the issue might have been fixed meanwhile! I will test this again as soon as our dev is back and active. Not huge issue right now tho

Sorry for the wait and thanks for your patience. Love what you are doing here at geyser! ❤️
